http-disk-server

```diff
@@ -320,12 +371,19 @@ def run_server(disk, ip, port):
     if not check_bindable(port):
         return
 
-    # Must be printed to notify parent processes that are waiting
-    # before starting the NBD server(s).
-    eprint('Server ready!')
-
-    # Note: `open_device` must be called after the message (never before!)
-    # to ensure we don't block if the device is already open somewhere.
+    # Note: We try to open the device and start the HTTP server to handle requests before
+    # sending the message below to the calling process. If we can't open the device, or if
+    # there is an issue during the first open call, we emit the message, because there
+    # is probably a running server on another machine.
```

Comment:
What if there's different issue with the device and all servers think there is another server running?

Reply:
The notification message will be sent, an we will retry to open the device as before this commit.
I don't see a better solution to handle this case now. Regarding the XCP-ng HA, it can fail the `xe pool-ha-enable` command, and in this case, we can suspect a problem with the device and which we probably cannot solve directly in this project.